gnuradio-runtime/lib/pmt/pmt_serialize.cc

```diff
@@ -26,25 +27,23 @@ static pmt_t parse_pair(std::streambuf& sb);
 // output primitives
 // ----------------------------------------------------------------
 
-static bool serialize_untagged_u8(unsigned int i, std::streambuf& sb)
+static bool serialize_untagged_u8(uint8_t i, std::streambuf& sb)
 {
     return sb.sputc((i >> 0) & 0xff) != std::streambuf::traits_type::eof();
 }
 
 // always writes big-endian
-static bool serialize_untagged_u16(unsigned int i, std::streambuf& sb)
+static bool serialize_untagged_u16(uint16_t i, std::streambuf& sb)
 {
-    sb.sputc((i >> 8) & 0xff);
-    return sb.sputc((i >> 0) & 0xff) != std::streambuf::traits_type::eof();
+    i = htobe16(i);
```

Comment:
In the sense of "include what you use", `#include <endian.h>` (I wonder what transitively includes that) would be right!